src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -174,6 +180,51 @@ class HttpGoogleServicesDAO(implicit val system: ActorSystem, implicit val mater
     scala.io.Source.fromInputStream(is).mkString
   }
 
+  /**
+    * Uploads the supplied data to GCS, using the Rawls service account credentials
+    * @param bucketName target bucket name for upload
+    * @param objectKey target object name for upload
+    * @param objectContents byte array of the data to upload
+    * @return path to the uploaded GCS object
+    */
+  override def writeObjectAsRawlsSA(bucketName: GcsBucketName, objectKey: GcsObjectName, objectContents: Array[Byte]): GcsPath = {
+    // if other methods in this class use google2/need these implicits, consider moving to the class level
+    // instead of here at the method level
+    implicit val cs: ContextShift[IO] = IO.contextShift(executionContext)
+    implicit val t: Timer[IO] = IO.timer(executionContext)
+    implicit val logger: SelfAwareStructuredLogger[IO] = Slf4jLogger.getLogger[IO]
+    val blocker = Blocker.liftExecutionContext(executionContext)
+
+    // create the storage service, using the Rawls SA credentials
+    // the Rawls SA json creds do not contain a project, so also specify the project explicitly
+    val storageResource = GoogleStorageService.resource(FireCloudConfig.Auth.rawlsSAJsonFile, blocker,
+       project = Some(GoogleProject(FireCloudConfig.FireCloud.serviceProject)))
+
+    // call the upload implementation
+    streamUploadObject(storageResource, bucketName, objectKey, objectContents)
+  }
+
+  // separate method to perform the upload, to ease unit testing
+  protected[dataaccess] def streamUploadObject(storageResource: Resource[IO, GoogleStorageService[IO]], bucketName: GcsBucketName,
+                                   objectKey: GcsObjectName, objectContents: Array[Byte]): GcsPath = {
+    // create the stream of data to upload
+    val dataStream: Stream[IO, Byte] = Stream.emits(objectContents).covary[IO]
+
+    val uploadAttempt = storageResource.use { storageService =>
+      // create the destination pipe to which we will write the file
+      // N.B. workbench-libs' streamUploadBlob does not allow setting the Content-Type, so we don't set it
+      val uploadPipe = storageService.streamUploadBlob(bucketName, GcsBlobName(objectKey.value))
+      // stream the data to the destination pipe
+      dataStream.through(uploadPipe).compile.lastOrError
+    }
+
+    // execute the upload
+    uploadAttempt.unsafeRunSync()
```

Comment:
I'm surprised we don't have existing scala code for writing a byte array to a bucket :thinking: